### ISSUE TITLE
Use epics-base instead of epicscorelibs

### ISSUE THREAD

**Issue by duncanmmacleod** [conda-forge/pyepics-feedstock#19]
I'm wondering whether it would be possible to switch out the dependency of this package to `epics-base`, rather than `epicscorelibs`. I've read through #17, and see that there are concerns that the `epics-base` package is broken on windows.

If we can resolve the epics-on-windows issues, is it reasonable to switch out the dependency, so that we can try and converge on `epics-base` as the one-true source of EPICS? I'm not an epics expert, just trying to simplify the packaging of `pyepics`, `pcaspy`, and a few other things needed by the LIGO experiment.

cc @jrollins.

**Comment by jrollins** [comment]
On Wed, Nov 20 2019, Duncan Macleod <notifications@github.com> wrote:
> I'm wondering whether it would be possible to switch out the
> dependency of this package to `epics-base`, rather than
> `epicscorelibs`. I've read through #17, and see that there are
> concerns that the `epics-base` package is broken on windows.
>
> If we can resolve the epics-on-windows issues, is it reasonable to
> switch out the dependency, so that we can try and converge on
> `epics-base` as the one-true source of EPICS? I'm not an epics expert,
> just trying to simplify the packaging of `pyepics`, `pcaspy`, and a
> few other things needed by the LIGO experiment.

For what it's worth, we have applications that use both pcaspy and
pyepics, so we need both of those packages to be compiled against the
same version of epics.

**Comment by hhslepicka** [comment]
@duncanmmacleod if you can work on the windows build for `epics-base` and get it going that would be great. 
I have an open issue (https://github.com/conda-forge/epics-base-feedstock/issues/3) and a PR that updates epics-base to 7.0.3 (https://github.com/conda-forge/epics-base-feedstock/pull/2).
I can help with tests if needed.
Also, if you do have `epics-base` installed, the `epicscorelibs` is smart enough to ignore itself and use your `libca` if `PYEPICS_LIBCA` is defined. It will not interfere with the `epics-base` package that you may already have installed and pyepics is not compiled so there is no issue in mixing it with pcaspy.

**Comment by beenje** [comment]
@hhslepicka, now that https://github.com/conda-forge/epics-base-feedstock/pull/6 was merged, are you interested in using `epics-base` as a dependency instead of `epicscorelibs`?
I can make a PR if you are.

**Comment by hhslepicka** [comment]
I will test epics-base here and get back to you on that.

**Comment by hhslepicka** [comment]
@beenje I think that it would be okay to do that. Just one request... Would you mind to only set the `PYEPICS_LIBCA` only if one is not set? That would allow for facilities that have their own EPICS Base compiled to also use that even if PyEPICS was installed via conda.

**Comment by beenje** [comment]
If we only set `PYEPCIS_LIBCA` when not already defined, it means we can't unset it in the deactivate script. That's not clean. Not speaking about the fact that the variable could have been set for different reasons and you might end-up with a conda environment not working as expected.
My opinion is that if this package depends on epics-base (from conda) it should use the libraries installed with conda.

conda is meant to install packages and their dependencies in an isolated environment. I don't really see why you would install pyepics with conda if you want to use another epics-base.
If that's what you want, isn't it better to install pyepics with pip (can still be done in a conda env)?
Another way would be to have a separate conda pyepics package that doesn't depend on epics-base.

Anyway I created #21 as I see it.
We can take the discussion there.
